pkg/cloud/tags.go

```diff
@@ -0,0 +1,63 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cloud
+
+type TagIFace interface {
+	AddNetworkTags(string, map[string]string) error
+	GetNetworkTags(string) (map[string]string, error)
+	DeleteNetworkTags(string, map[string]string) error
+}
+
+const (
+	clusterTagNamePrefix = "CAPC_cluster_"
+	createdByCapcTagName = "created_by_CAPC"
+	resourceTypeNetwork  = "network"
+)
+
+// TagNetwork adds tags to a network by network id.
+func (c *client) AddNetworkTags(networkId string, tags map[string]string) error {
+	// https://cloudstack.apache.org/api/apidocs-4.16/apis/createTags.html
```

Comment:
Let's go ahead and remove the links to the api documentation here. I just put them in for reference when you were getting started. : )